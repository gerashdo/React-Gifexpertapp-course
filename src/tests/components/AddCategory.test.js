import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";


describe('Pruebas del componente AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    
    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
    });

    test('Debe de renderizarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate( 'change' , { 
            target: { value }
        });
        
    })

    test('no debe de postear la informacion on submit', () => {
       
        wrapper.find( 'form' ).simulate( 'submit' , { preventDefault(){} } );

        expect( setCategories ).not.toHaveBeenCalled();

    });
    
    


    

});