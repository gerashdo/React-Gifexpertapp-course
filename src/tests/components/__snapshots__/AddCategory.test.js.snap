// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente AddCategory Debe de renderizarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
